contracts/Karmic.sol

```diff
@@ -32,16 +69,31 @@ contract Karmic is Badger{
         }
     }
 
+    function withdraw(address token, uint256 amount) external{
+        require(!(boxTokenTiers[token].funds >= boxTokenTiers[token].threshold),
+            "Can withdraw only funds for tokens that didn't pass threshold");
+        require(boxTokenTiers[token].id != 0,
+            "Can withdraw only funds for crowdfund tokens");
+        uint256 withdrawnFunds = amount * boxTokenTiers[token].funds / boxTokenTiers[token].amount;
+        require(ERC20(token).transferFrom(msg.sender, address(this), amount),
+            "Failed to withdraw stakeholder's ERC20 tokens");
+        payable(msg.sender).transfer(withdrawnFunds);
+    }
+
+
     function claimGovernanceTokens(address[] memory boxTokens) external {
         bytes memory data;
 
         address token;
         for(uint8 i; i < boxTokens.length; i++) {
             token = boxTokens[i];
             uint256 amount = IERC20(token).balanceOf(msg.sender);
-            uint256 tokenId = boxTokenTiers[token];
+            boxTokenTiers[token].amount -= amount;
+            uint256 tokenId = boxTokenTiers[token].id;
             IERC20(token).transferFrom(msg.sender, address(this), amount);
```

Comment:
I just noticed that I forgot to check the return value of this transfer operation. Right now, it might be possible to claim gov tokens without passing box tokens 🙈  Could you fix that one for me? 🐰

Reply:
Did it